components/advanced/babel-plugin.js

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import SectionLayout from '../SectionLayout';
+import CodeBlock from '../CodeBlock';
+import Code from '../Code';
+import Note from '../Note';
+
+const installNPM = `
+npm install --save-dev babel-plugin-styled-components
+`.trim();
+
+const usage = `
+{
+  "plugins": ["styled-components"]
+}
+`.trim();
+
+const ssr = `
+{
+  "plugins": [
+    ["styled-components", {
+      "ssr": true
+    }]
+  ]
+}
+`.trim();
+
+const displayName = `
+{
+  "plugins": [
+    ["styled-components", {
+      "displayName": false
+    }]
+  ]
+}
+`.trim();
+
+const preprocess = `
+{
+  "plugins": [
+    ["styled-components", {
+      "preprocess": true
+    }]
+  ]
+}
+`.trim();
+
+const minify = `
+{
+  "plugins": [
+    ["styled-components", {
+      "minify": false
+    }]
+  ]
+}
+`.trim();
+
+const BabelPlugin = () =>
+  <SectionLayout title="Babel Plugin" labels={['v2']}>
+    <p>
+      This plugin adds support for server-side rendering, for minification of
+      styles and gives you a nicer debugging experience when using{' '}
+      <Code>styled-components</Code>.
+    </p>
+    <p>Usage</p>
+    <CodeBlock code={installNPM} language="node" />
+    <p>
+      Then in your babel configuration (probably <Code>.babelrc</Code>):
+    </p>
+    <CodeBlock code={usage} language="node" />
+    <SectionLayout sub title="Server-side rendering">
+      <Note>This option is turned off by default</Note>
+      <p>
+        By adding a unique identifier to every styled component this plugin
+        avoids checksum mismatches due to different class generation on the
+        client and on the server. If you do not use this plugin and try to
+        server-side render <Code>styled-components</Code> React will complain.
```

Comment:
`<Code>styled-components</Code>` needs to be changed to just `styled components`. We're not talking about the lib :smile:

> If you do not use this plugin and try to server-side render styled-components React will complain.

This is a bit outdated and also sounds weird, so let's update it:

> If you do not use this plugin and try to server-side render, you will likely encounter React warnings, if the import order is different on the client. For this reason we generally recommend you to turn on this option in any SSR setup.

We also need to add a reference to this babel-plugin option to our SSR section.